### ISSUE TITLE
Reported size are not (always) correct

### ISSUE THREAD

**Issue by theKashey** [ai/size-limit#121]
Comparing `0.21` vs `1.3`:

### 0.21 - that's a true size
```text
 dist/es5/index.js
  Package size: 7.91 KB

  dist/es2015/sidecar.js
  Package size: 4.31 KB

  dist/es2015/UI.js
  Package size: 2.45 KB
```

### 1.3.6
```text
  dist/es5/index.js
  Package size: 1.47 KB

  dist/es2015/sidecar.js
  Package size: 807 B

  dist/es2015/UI.js
  Package size: 576 B
```

## Another project
Which is __included__ in the project above, ie `sidecar.js` == `project1car.js + project2car.js`.

### 0.21
```text
 dist/cjs/UI.js
  Package size: 2.5 KB

  dist/es2015/sidecar.js
  Package size: 2.95 KB

  dist/es2015/index.js
  Package size: 4.33 KB
```

### 1.3.6
```text
 dist/cjs/UI.js
  Package size: 1.69 KB

  dist/es2015/sidecar.js
  Package size: 2.63 KB

  dist/es2015/index.js
  Package size: 3.79 KB
```
In this case, the difference is not as big, but still present. Probably that's some webpack optimizations, but size-limit has no fixed webpack version set, so it shall be the same between different `size-limit` versions.

---

According to `size-limit` side-car of the following project is about 3kb, while sidecar of a parent project(which imports it), is just 807b. Meanwhile, 0.21 reported the size correctly.

According to `--why` 2 of 4 external packages do not "present", even in a `dev` mode, that's why size is so small - it calculates nothing.

I have no idea about why this is happening, but have to report the issue.

**Comment by ai** [comment]
Do you have any `peerDependencies` in `package.json` or `ignore` in Size Limit config?

**Comment by theKashey** [comment]
I've disabled everything, including temporal bundle removal, and now I am trying to understand the way the bundle is built. And that's super interesting - look like the module resolution is completely broken in my case.

**Comment by ai** [comment]
Can you try `--output-bundle build/`?

**Comment by theKashey** [comment]
Given a code
```js
import * as React from 'react';
import { hideOthers } from 'aria-hidden';
import { InteractivityDisabler } from "./InteractivityDisabler";
import { focusHiddenMarker } from "./medium";
```
Result
```js
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! react */ "react-focus-lock");
!!!! ^^^^^^ !!!!!
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var aria_hidden__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! aria-hidden */ "./node_modules/aria-hidden/dist/es2015/index.js");
/* harmony import */ var _InteractivityDisabler__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ./InteractivityDisabler */ "./dist/es2015/InteractivityDisabler.js");
/* harmony import */ var _medium__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! ./medium */ "./dist/es2015/medium.js");
```

Given a code 
```js
import { exportSidecar } from "use-sidecar";
import 'react-focus-lock';
import 'react-remove-scroll';
import { Effect } from './Effect';
import { effectCar } from "./medium";
```
Result 
```js
/* harmony import */ var use_sidecar__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! use-sidecar */ "./node_modules/use-sidecar/dist/es2015/index.js");
/* harmony import */ var react_focus_lock__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! react-focus-lock */ "react-focus-lock");
!!!! ^^^^^^, and there is no second import !!!!!
/* harmony import */ var react_focus_lock__WEBPACK_IMPORTED_MODULE_1___default = /*#__PURE__*/__webpack_require__.n(react_focus_lock__WEBPACK_IMPORTED_MODULE_1__);
/* harmony import */ var _Effect__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ./Effect */ "./dist/es2015/Effect.js");
/* harmony import */ var _medium__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! ./medium */ "./dist/es2015/medium.js");
```

The same file build by local dev server
```js
var use_sidecar_1 = __webpack_require__(/*! use-sidecar */ "./node_modules/use-sidecar/dist/es2015/index.js");
__webpack_require__(/*! react-focus-lock/sidecar */ "./node_modules/react-focus-lock/dist/es2015/sidecar.js");
__webpack_require__(/*! react-remove-scroll/sidecar */ "./node_modules/react-remove-scroll/dist/es2015/sidecar.js");
var Effect_1 = __webpack_require__(/*! ./Effect */ "./src/Effect.tsx");
var medium_1 = __webpack_require__(/*! ./medium */ "./src/medium.ts");
```

**Comment by theKashey** [comment]
I've __found the issue__ - it's `opts.ignore`, which set to `['react']` (however I did not specified it).
Result - it ignores all the packages __starting from "react"__ (`if (opts.ignore.some(i => request.startsWith(i))) {`), and that's in my case - ALL THE PACKAGES.

**Comment by theKashey** [comment]
https://github.com/ai/size-limit/blob/master/index.js#L102 is unsafe operation and should be reconsidered.
Probably it should be ```if (opts.ignore.some(i => request.startsWith(`${i}/`))) {```, ie with a trailing slash.

**Comment by ai** [comment]
Released in 1.3.7
